sources/dicom/large_image_source_dicom/__init__.py

```diff
@@ -143,6 +148,87 @@ def __init__(self, path, **kwargs):
             max(self.sizeX / self.tileWidth, self.sizeY / self.tileHeight)) / math.log(2)) + 1))
         self._populatedLevels = len(self._dicom.levels)
 
+    def _open_wsi_dicom(self, path):
+        if isinstance(path, dict):
+            # Use the DICOMweb open method
+            return self._open_wsi_dicomweb(path)
+        else:
+            # Use the regular open method
+            return wsidicom.WsiDicom.open(path)
+
+    def _open_wsi_dicomweb(self, info):
+        # These are the required keys in the info dict
+        url = info['url']
+        study_uid = info['study_uid']
+        series_uid = info['series_uid']
+
+        # These are optional keys
+        qido_prefix = info.get('qido_prefix')
+        wado_prefix = info.get('wado_prefix')
+        auth = info.get('auth')
+
+        # Create the client
+        client = wsidicom.WsiDicomWebClient(
+            url,
+            qido_prefix=qido_prefix,
+            wado_prefix=wado_prefix,
+            auth=auth,
+        )
+
+        # Identify the transfer syntax
+        transfer_syntax = self._identify_dicomweb_transfer_syntax(client,
+                                                                  study_uid,
+                                                                  series_uid)
+
+        # Open the WSI DICOMweb file
+        return wsidicom.WsiDicom.open_web(client, study_uid, series_uid,
+                                          requested_transfer_syntax=transfer_syntax)
+
+    def _identify_dicomweb_transfer_syntax(self, client, study_uid, series_uid):
+        # "client" is a wsidicom.WsiDicomWebClient
+
+        # This is how we select the JPEG type to return
+        # The available transfer syntaxes used by wsidicom may be found here:
+        # https://github.com/imi-bigpicture/wsidicom/blob/a2716cd6a443f4102e66e35bbce32b0e2ae72dab/wsidicom/web/wsidicom_web_client.py#L97-L109
+        # (we may need to update this if they add more options)
+        # FIXME: maybe this function better belongs upstream in `wsidicom`?
+        from pydicom.uid import JPEG2000, JPEG2000Lossless, JPEGBaseline8Bit, JPEGExtended12Bit
+
+        # Prefer the transfer syntaxes in this order.
+        transfer_syntax_preferred_order = [
+            JPEGBaseline8Bit,
```

Comment:
I notice that some of the dicoms on the webserver we were looking at have different transfer syntaxes (like JPEG-LS variants).  What happens if we do `transfer_syntax_preferred_order = pydicom.uid.AllTransferSyntaxes`?  Do not all of them work, or is the order bad?  This could be investigated in a future PR.

Reply:
Well, our dependency `wsidicom` only supports a subset of those right now. For example see [here](https://github.com/imi-bigpicture/wsidicom/blob/9a7a33714823be42b830405a955383f90cb68d1e/wsidicom/web/wsidicom_web_client.py#L97-L109). I think we would encounter an error in `wsidicom` if we did not use one of those. Also see this [note in their README](https://github.com/imi-bigpicture/wsidicom#limitations).

I'm not sure if they would be able to support JPEG-LS, but we could perhaps ask!

Also, this function might belong better in `wsidicom` rather than here. We are simply making a REST request to see what type of format it supports, and then asking for that format.

Reply:
Great.  I'd like to make an issue for more transfer format support on our repo; we can figure out how to contribute upstream or handle it ourselves when we get to it.